[DX-1008] Updated Internal Endpoint middleware (#4046)
TykTechnologies/tyk-docs: tyk-docs/content/product-stack/tyk-gateway/middleware/internal-endpoint-tyk-classic.md

---
title: Using the Internal Endpoint middleware with Tyk Classic APIs
date: 2024-01-26
description: "Using the Internal Endpoint middleware with Tyk Classic APIs"
tags: ["internal endpoint", "internal", "middleware", "per-endpoint", "Tyk Classic"]
---

The [Internal Endpoint]({{< ref "product-stack/tyk-gateway/middleware/internal-endpoint-middleware" >}}) middleware instructs Tyk Gateway not to process external requests to the endpoint (which is a combination of HTTP method and path). Internal requests from other APIs will be processed.

> When working with Tyk Classic APIs, the middleware is configured in the Tyk Classic API Definition either manually within the `.json` file or from the API Designer in the Tyk Dashboard.

Author: ```suggestion
When working with Tyk Classic APIs, the middleware is configured in the Tyk Classic API Definition. You can do this via the Tyk Dashboard API or in the API Designer.
```

## Configuring the middleware in the Tyk Classic API Definition
To enable the middleware you must add a new `internal` object to the `extended_paths` section of your API definition.

The `internal` object has the following configuration:
 - `path`: the path to match on
 - `method`: this method to match on

For example:
```.json  {linenos=true, linenostart=1}
{
    "extended_paths": {
        "internal": [
            {
                "disabled": false,
                "path": "/status/200",
                "method": "GET"
            }
        ]
    }
}
```

In this example the internal endpoint middleware has been configured for HTTP `GET` requests to the `/status/200` endpoint. Any calls made to this endpoint that originate outside Tyk (i.e. externally) will be rejected with `HTTP 403 Forbidden`, however it can be reached internally by another API at `tyk://<listen_path>/status/200`.

## Configuring the middleware in the API Designer
You can use the API Designer in the Tyk Dashboard to configure the internal endpoint middleware for your Tyk Classic API by following these steps.

> #### Step 1: Add an endpoint for the path and select the plugin

Author: I would want to put emphasis on the word `internal` here because it's a common word - but that's tricky as it's in the title... hence Ieft it out.
